internal/provider/data_source_http.go

```diff
@@ -252,6 +270,17 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		clonedTr.TLSClientConfig.RootCAs = caCertPool
 	}
 
+	if !model.ClientCert.IsNull() && !model.ClientKey.IsNull() {
+		cert, err := tls.X509KeyPair([]byte(model.ClientCert.ValueString()), []byte(model.ClientKey.ValueString()))
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"error creating x509 key pair",
+				fmt.Sprintf("error creating x509 key pair from provided pem blocks\n\nError: %s", err),
+			)
```

Comment:
I think we should early return on error here
```suggestion
			)
			return
```